### PR TITLE
[RHELC-903] Remove unused dependency error resolution functions

### PR DESCRIPTION
**Pull request by r0x0d**
Some dependencies related to dependency error resolution were left behind while working on https://github.com/oamg/convert2rhel/pull/669.

This commit introduces the removal of the functions that were being left out.

Jira Issues: [RHELC-903](https://issues.redhat.com/browse/RHELC-903)

Checklist

- [ ] PR has been tested manually in a VM (either author or reviewer)
- [x] Jira issue has been made public if possible
- [x] `[RHELC-]` is part of the PR title 
- [ ] GitHub label has been added to help with Release notes 
- [ ] PR title explains the change from the user's point of view
- [ ] Code and tests are documented properly
- [ ] The commits are squashed to as few commits as possible (without losing data) 
- [ ] When merged: Jira issue has been updated to `Release Pending` if relevant

### CODE REVIEW

**Comment by r0x0d** [comment]
@bocekm I removed a few extra functions that were used internally for the ones that were described in the ticket to be removed.

File: convert2rhel/unit_tests/pkghandler_test.py [hunk 1]
```diff
@@ -318,54 +318,6 @@ def test_call_yum_cmd_with_repo_overrides(self):
...
-            | problematic_pkgs["protected"]
-            | problematic_pkgs["multilib"]
-            | problematic_pkgs["required"],
-        )
-
-    @unit_tests.mock(pkghandler, "call_yum_cmd", CallYumCmdMocked())
-    def test_resolve_dep_errors_one_downgrade_fixes_the_error(self):
-        pkghandler.call_yum_cmd.fail_once = True
-
-        pkghandler.resolve_dep_errors(YUM_PROTECTED_ERROR, set())
```
Comment (bocekm) [inline convert2rhel/unit_tests/pkghandler_test.py:342]: You can remove all the constants used by the removed tests on lines [1741-1843](https://github.com/oamg/convert2rhel/pull/822/files#diff-c3ce6b0e1e7ddae0678f9eb7c71f9c5f8aecbfd5b7db1deb7d8947ccf1492a08L1741-L1843).

Comment (r0x0d) [reply convert2rhel/unit_tests/pkghandler_test.py:342]: good catch! will remove

File: convert2rhel/unit_tests/pkghandler_test.py [hunk 2]
```diff
@@ -1843,210 +1717,6 @@ def test_get_packages_to_update_dnf(packages, reposdir, pretend_os, monkeypatch)
     YUM_DISTRO_SYNC_OUTPUT = f.read()
```
Comment (bocekm) [inline convert2rhel/unit_tests/pkghandler_test.py:1717]: This one can be removed as well.

**Review by bocekm — CHANGES_REQUESTED** [review]

File: convert2rhel/unit_tests/pkghandler_test.py (hunk 2 above)
Comment (r0x0d) [reply convert2rhel/unit_tests/pkghandler_test.py]: done.

File: convert2rhel/unit_tests/pkghandler_test.py [hunk 3]
```diff
@@ -426,53 +426,39 @@ def test_call_yum_cmd_with_repo_overrides(self):
...
-
-        pkghandler.resolve_dep_errors(YUM_PROTECTED_ERROR, set())
-
-        # Firts call of the resolve_dep_errors, pkgs from protected error
-        # are detected, the second call pkgs from multilib error are detected,
-        # the third call yum_cmd is not called anymore, because the
-        # problematic packages then remain the same (simulating that the
-        # downgrades do not solve the yum errors)
-        self.assertEqual(pkghandler.call_yum_cmd.called, 2)
+    class GetInstalledPkgsWFingerprintsMocked(unit_tests.MockFunction):
```
Comment (bocekm) [inline convert2rhel/unit_tests/pkghandler_test.py:429]: This seems to me as an accidental revert of changes from https://github.com/oamg/convert2rhel/commit/5666d74604582665281194819cbfb54131747073#diff-c3ce6b0e1e7ddae0678f9eb7c71f9c5f8aecbfd5b7db1deb7d8947ccf1492a08L369.

File: convert2rhel/unit_tests/pkghandler_test.py [hunk 4]
```diff
@@ -175,22 +175,6 @@ def __call__(self, pkgs):
         self.pkgs = pkgs
 
 
-class GetInstalledPkgObjectsMocked(unit_tests.MockFunction):
```
Comment (r0x0d) [inline convert2rhel/unit_tests/pkghandler_test.py:178]: Was not unused by any tests anymore

File: convert2rhel/unit_tests/pkghandler_test.py (hunk 3 above)
Comment (r0x0d) [reply convert2rhel/unit_tests/pkghandler_test.py]: yeah, it was an accidental revert caused by the rebase. It is not there anymore.

**Comment by r0x0d** [comment]
Waiting for the integration tests to finish so I can merge it.

**Pull request merged**